File: pyproject1/translate.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "scanner.h"
#include "support.h"
int main(int argc,char **argv)
       {
       int dicount;//holds the number for the dictionary
       int msgcount;//holds the number for the messages
       char ***dic;//holds the array of arrays for the dictionary
       char **msg;//holds the array for the messages
       //checks the number of arguments and exits the program if not met
       if (argc != 3)
            {
            printf("need two arguments!\n");
            exit(1);
            }
        //count and fill arrays
        dicount = countDic(argv[1]);
        dic = readTable(argv[1]);
        msgcount = countTokens(argv[2]);
        msg = readMsg(argv[2]);
        //translate
        char * WorthlessPrint;
        int i;
        
        for (i = 0; i < msgcount; ++i)
        {
          
          WorthlessPrint = translate(msg[i],dic,dicount);
       
          if (WorthlessPrint == 0)
          {
            printf("%s ",msg[i] );
          }

          else 
          {
            printf("%s ",WorthlessPrint );
          }
         
          

        }
        printf("\n");
       
        //free arrays
        freeTable(dic,dicount);
        freeArray(msg,msgcount);
        return 0;
       }

File: pyproject1/translate.py
from scanner import *
from support import *
import sys


def main():
    if len(sys.argv) != 3:
        print("need two argvs")
        sys.exit(0)
    dicount = countDic(sys.argv[1])
    dic = readArray(sys.argv[1], dicount)
    msgcount = countTokens(sys.argv[2])
    msg = readMsg(sys.argv[2], msgcount)
    print("", msg[msgcount - 1])
    print("", dic[0])
    for i in range(0, msgcount):
        translateable = translate(msg[i], dic, dicount)
        if translateable == 0:
            print(msg[i], "", end="")
        else:
            print(translateable, "", end="")
    print("\n")


main()
